python_modules/dagster/dagster/_scheduler/scheduler.py

```diff
@@ -483,6 +490,24 @@ def launch_scheduled_runs_for_schedule_iterator(
     if not tick_times:
         if log_verbose_checks:
             logger.info(f"No new tick times to evaluate for {schedule_name}")
+
+        # insert last considered timestamp here
+        if (
+            not instigator_data.last_iteration_timestamp
+            or instigator_data.last_iteration_timestamp + LAST_RECORDED_ITERATION_INTERVAL
```

Comment:
seems like theres an assumption being taken here 

instead of keeping this totally up to date, we keep it up to date within the tolerance of `LAST_RECORDED_ITERATION_INTERVAL` which we assume any deletes will be greater than that time

we could consider making it invalid to set the bit of instance config that cleans up ticks to less than this value (or some relative math) 

short of that we should leave some comments explaining whatsup

Does this window have an implication on schedules changes that are faster than this ?

Reply:
> Does this window have an implication on schedules changes that are faster than this ?
Let's say I have a schedule that fires on Sundays at 12:00 AM.  On Wednesday at 2:00 AM, I change this schedule to fire on Wednesday at 1:30 AM and deploy it, we may kick off a run for the 1:30 AM tick.  It's somewhat non-deterministic because it depends on the logged iteration of the scheduler, but also it's not that crazy unexpected to see a run for the previous tick if it lands within the hour.  

This is a consideration independent of tick deletions, since we can have the above scenario, without any ticks logged (let's say that I turned the schedule on Monday morning).

Reply:
Re: tick deletion, initially had this stacked on https://github.com/dagster-io/dagster/pull/14555, but decoupled these two so that we could merge this fix and solve the tick deletion issue a different way.